hera_cal/lstbin_simple.py

```diff
@@ -0,0 +1,1677 @@
+
+        bls = np.array([antpos[k[0]] - antpos[k[1]] for k in baselines])
+
+        # get appropriate lst_shift for each integration, then rephase
+        lst_shift = lst_bin_centres[grid_indices] - data_lsts
+
+        # this makes a copy of the data in d
+        utils.lst_rephase_vectorized(data, bls, freq_array, lst_shift, lat=lat, inplace=True)
+
+    # shortcut -- just return all the data, re-organized.
+    _data, _flags, _nsamples = [], [], []
+    empty_shape = (0, len(baselines), len(freq_array), npols)
+    for lstbin in range(len(lst_bin_centres)):
+        mask = grid_indices == lstbin
+        if np.any(mask):
+            _data.append(data[mask])
+            _flags.append(flags[mask])
+            _nsamples.append(nsamples[mask])
+        else:
+            _data.append(np.zeros(empty_shape, complex))
+            _flags.append(np.zeros(empty_shape, bool))
+            _nsamples.append(np.zeros(empty_shape, int))
+
+    return lst_bin_centres, _data, _flags, _nsamples
+
+def get_lst_bins(lsts: np.ndarray, edges: np.ndarray) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Get the LST bin indices for a set of LSTs.
+    
+    Parameters
+    ----------
+    lsts
+        The LSTs to bin, in radians.
+    edges
+        The edges of the LST bins, in radians.
+
+    Returns
+    -------
+    bins
+        The bin indices for each LST.
+    lsts
+        The LSTs, wrapped so that the minimum is at the lowest edge, and all are within
+        2pi of that minimum.
+    mask
+        A boolean mask indicating which LSTs are within the range of the LST bins.
+    """
+    lsts = np.array(lsts).copy()
+    
+    # Now ensure that all the observed LSTs are wrapped so they start above the first bin edges
+    lsts %= 2*np.pi
+    lsts[lsts < edges[0]] += 2* np.pi
+    bins = np.digitize(lsts, edges, right=True) - 1
+    mask = (bins >= 0) & (bins < (len(edges)-1))
+    return bins, lsts, mask
+
+def reduce_lst_bins(
+    data: list[np.ndarray], flags: list[np.ndarray], nsamples: list[np.ndarray],
+    out_data: np.ndarray | None = None, 
+    out_flags: np.ndarray | None = None,
+    out_std: np.ndarray | None = None,
+    out_nsamples: np.ndarray | None = None,
+    mutable: bool = False,
+    sigma_clip_thresh: float = 0.0,
+    sigma_clip_min_N: int = 4,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    From a list of LST-binned data, produce reduced statistics.
+
+    Use this function to reduce lists of arrays with multiple time integrations per bin
+    (i.e. the output of :func:`simple_lst_bin`) to arrays of shape 
+    ``(nbl, nlst_bins, nfreq, npol)``. For example, compute the mean/std.
+
+    Parameters
+    ----------
+    data
+        The data to perform the reduction over. The length of the list is the number
+        of LST bins. Each array in the list should have shape 
+        ``(nbl, ntimes_per_lst, nfreq, npol)``.
```

Comment:
(nbl, nsamples_per_lst, nfreq, npol)

Reply:
I've used `nintegrations_in_lst`. What do you think?

Reply:
Again, I've used `nintegrations_in_lst` instead of `ntimes_per_lst`. Lmk what you think